sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManager.kt

```diff
@@ -0,0 +1,135 @@
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.Geometry
+import com.mapbox.geojson.Point
+import com.mapbox.maps.MapView
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.ViewAnnotationOptions
+
+/**
+ * Manager API to control View Annotations.
+ *
+ * View annotations are Android [View]s that are drawn on top of the [MapView] and bound to some [Geometry] (only [Point] is supported for now).
+ * In case some view annotations intersect on the screen Z-index is based on addition order.
+ *
+ * View annotations are invariant to map camera transformations however such properties as size, visibility etc
+ * could be controlled by the user using update operation.
+ *
+ * View annotations are not explicitly bound to any sources however [ViewAnnotationOptions.associatedFeatureId] could be
+ * used to bind given view annotation with some [Feature] by [Feature.id] meaning visibility of view annotation will be driven
+ * by visibility of given feature.
+ */
+@MapboxExperimental
+interface ViewAnnotationManager {
+
+  /**
+   * Add view annotation inflated from [resId] synchronously.
+   * Parent layout of [resId] must have fixed dimensions and should not use [ViewGroup.LayoutParams.WRAP_CONTENT].
+   *
+   * Annotation [options] must include Geometry where we want to bind our view annotation.
+   *
+   * Width and height could be specified explicitly but better idea will be not specifying them
+   * as they will be calculated automatically based on view layout.
+   *
+   * @param resId layout resource id
+   * @param options view annotation options
+   *
+   * @return inflated [View].
+   *
+   * @throws [RuntimeException] if options did not include geometry.
+   */
+  fun addViewAnnotation(
+    @LayoutRes resId: Int,
+    options: ViewAnnotationOptions
+  ): View
+
+  /**
+   * Add view annotation inflated from [resId] asynchronously.
+   * Parent layout of [resId] must have fixed dimensions and should not use [ViewGroup.LayoutParams.WRAP_CONTENT].
+   *
+   * In order to use this function please add following [dependency](https://mvnrepository.com/artifact/androidx.asynclayoutinflater/asynclayoutinflater/1.0.0) to your project.
+   *
+   * Annotation [options] must include Geometry where we want to bind our view annotation.
+   *
+   * Width and height could be specified explicitly but better idea will be not specifying them
+   * as they will be calculated automatically based on view layout.
+   *
+   * @param resId layout resource id
+   * @param options view annotation options
+   * @param asyncInflater instance of [AsyncLayoutInflater] provided by the user
+   * @param asyncInflateCallback callback triggered when [View] is inflated.
+   *
+   * @throws [RuntimeException] if options did not include geometry.
+   */
+  fun addViewAnnotation(
+    @LayoutRes resId: Int,
+    options: ViewAnnotationOptions,
+    asyncInflater: AsyncLayoutInflater,
```

Comment:
seems the asyncInflater is required to use the view annotations, why did we use compileOnly dependency on it?

Reply:
see https://github.com/mapbox/mapbox-maps-android/pull/834#discussion_r750518789